src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/OrchestratorMethodCollector.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Analyzers
+{
+    public class OrchestratorMethodCollector
+    {
+        private Dictionary<ISymbol, MethodInformation> orchestratorMethodDeclarations = new Dictionary<ISymbol, MethodInformation>();
+
+        public void FindOrchestratorMethods(SyntaxNodeAnalysisContext context)
+        {
+            var semanticModel = context.SemanticModel;
+            var symbol = context.ContainingSymbol;
+            if (context.Node is MethodDeclarationSyntax declaration
+                && SyntaxNodeUtils.IsInsideOrchestratorFunction(semanticModel, declaration)
+                && symbol != null)
```

Comment:
Consider moving the `symbol != null` check earlier in this if-statement since it is less CPU intensive compared to `IsInsideOrchestratorFunction`.